plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -1143,32 +1144,23 @@ public void verifySupportsJoinPushdownDeclaration()
         }
 
         assertThat(query(joinPushdownEnabled(getSession()), "SELECT r.name, n.name FROM nation n JOIN region r ON n.regionkey = r.regionkey"))
-                .isNotFullyPushedDown(
-                        node(JoinNode.class,
-                                anyTree(node(TableScanNode.class)),
-                                anyTree(node(TableScanNode.class))));
+                .joinIsNotFullyPushedDown();
     }
 
-    @Test
-    public void testJoinPushdown()
+    @Test(dataProvider = "joinOperators")
+    public void testJoinPushdown(JoinOperator joinOperator)
     {
-        PlanMatchPattern joinOverTableScans =
-                node(JoinNode.class,
-                        anyTree(node(TableScanNode.class)),
-                        anyTree(node(TableScanNode.class)));
-
-        PlanMatchPattern broadcastJoinOverTableScans =
-                node(JoinNode.class,
-                        node(TableScanNode.class),
-                        exchange(ExchangeNode.Scope.LOCAL,
-                                exchange(ExchangeNode.Scope.REMOTE, ExchangeNode.Type.REPLICATE,
-                                        node(TableScanNode.class))));
-
         Session session = joinPushdownEnabled(getSession());
 
         if (!hasBehavior(SUPPORTS_JOIN_PUSHDOWN)) {
             assertThat(query(session, "SELECT r.name, n.name FROM nation n JOIN region r ON n.regionkey = r.regionkey"))
-                    .isNotFullyPushedDown(joinOverTableScans);
+                    .joinIsNotFullyPushedDown();
+            return;
+        }
+
+        if (joinOperator == FULL_JOIN && !hasBehavior(SUPPORTS_JOIN_PUSHDOWN_WITH_FULL_JOIN)) {
+            assertThat(query(session, "SELECT r.name, n.name FROM nation n FULL JOIN region r ON n.regionkey = r.regionkey"))
+                    .joinIsNotFullyPushedDown();
             return;
         }
```

Comment:
Extract to a separate test method similar to `verifySupportsJoinPushdownDeclaration` or inline `verifySupportsJoinPushdownDeclaration`.

Reply:
Done